src/charm.py

```diff
@@ -167,6 +172,8 @@ def __init__(self, *args, **kargs) -> None:
         no_proxy = get_env_var("JUJU_CHARM_NO_PROXY")
         if (https_proxy or http_proxy) and no_proxy:
             self.proxies["no_proxy"] = no_proxy
+        if self._state.aproxy_address:
+            self.proxies["aproxy_address"] = self._state.aproxy_address
```

Comment:
No, not really. I suppose you'd just do them as charm config, perhaps with a special name/prefix to indicate it's a (short-lived) feature flag rather than stable config?

Reply:
Thank you, Ben, for suggesting this.